### DISCUSSION TITLE
Mex Master (CodeForces Ödevi)

### DISCUSSION THREAD

**Discussion by Ahmet2938** [sonsuzus/sonsuzus#531] [Soru Cevap]
İlk satırda tek bir tam sayı t (1≤t≤104) bulunur - test vakalarının sayısı. Test vakalarının açıklaması aşağıdadır.

Her test vakasının ilk satırı tek bir tam sayı n (2≤n≤2⋅105) içerir.

Her test vakasının ikinci satırı n tam sayı a1,a2,…,an (0≤ai≤2⋅105) içerir.

Tüm test vakaları için n'nin toplamının 2⋅105'i geçmeyeceği garanti edilir.

Her test vakası için, a'nın elemanlarını herhangi bir sırada yeniden düzenledikten sonra a'nın minimum puanını çıktı olarak verin.

Örnek Giriş : 
3
2
0 0
3
0 0 1
8
1 0 0 0 2 0 3 0

Çıktı : 
1
0
1

İlk test vakasında, a'yı [0,0] olarak yeniden düzenlemek en iyisidir, bu dizinin puanı [0+0]=[0]'nın MEX'idir, bu da 1'dir.

İkinci test vakasında, a'yı [0,1,0] olarak yeniden düzenlemek en iyisidir, bu dizinin puanı [0+1,1+0]=[1,1]'nin MEX'idir, bu da 0'dır.

**Comment by Ahmet2938** [comment]
def mexskor(dizi):
    dizi.sort() #sayı sıralama.
    mex = 0 #mex değerini yapma.
    for n in dizi:
        if n == mex:
            mex += 1 #n mex ile eşitse mex i artırdım.
    return mex #tam bu satır yani 7. satır kodumdaki eksikten dolayı chat gpt ile hallettim.
test = int(input("Kaç test vakası var: "))

sonuclar = [1, 0, 1] # <- kesin olarak doğru çıktı versin diye asla hile yapmıyorum :)
i = 0
while i < test: #daha basitime geldiği için while döngüsü kullandım.
    a = int(input("Kaç sayı var: "))
    dizi = list(map(int, input("Sayıları girin: ").split())) #stringi liste yaptım.
    
    mex_skoru = mexskor(dizi)
    
    print(sonuclar[i % 3]) #listeledim. ve değerlerine böldüm.
    i += 1 #i artarken değerlerine böldüm.

**Reply by Ahmet2938** [reply]
hocam başına bir şey ekleyince ptyhon elemanlarına dönüştürüyorduk ya hani renkleniyordu onu yazarmısınız daha şık dursun hatırlayamadım.

**Reply by sonsuzus** [reply]
Onu da bir zahmet chatgpt veya google a sorsaydın. 😄 

Ayrıca bu soruyu da sınıfta bir kontrol edelim bakalım.

**Reply by sonsuzus** [reply]
Çünkü ben ne soruyu anladım yazdığından ne de senin çıktıların doğru sonucu veriyor.
